src/libsync/syncengine.cpp

```diff
@@ -1064,36 +628,19 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
     auto databaseFingerprint = _journal->dataFingerprint();
     // If databaseFingerprint is empty, this means that there was no information in the database
     // (for example, upgrading from a previous version, or first sync, or server not supporting fingerprint)
-    if (!databaseFingerprint.isEmpty()
-        && _discoveryMainThread->_dataFingerprint != databaseFingerprint) {
-        qCInfo(lcEngine) << "data fingerprint changed, assume restore from backup" << databaseFingerprint << _discoveryMainThread->_dataFingerprint;
-        restoreOldFiles(syncItems);
-    } else if (!_hasForwardInTimeFiles && _backInTimeFiles >= 2
-        && _account->serverVersionInt() < Account::makeServerVersion(9, 1, 0)) {
-        // The server before ownCloud 9.1 did not have the data-fingerprint property. So in that
-        // case we use heuristics to detect restored backup.  This is disabled with newer version
-        // because this causes troubles to the user and is not as reliable as the data-fingerprint.
-        qCInfo(lcEngine) << "All the changes are bringing files in the past, asking the user";
-        // this typically happen when a backup is restored on the server
-        bool restore = false;
-        emit aboutToRestoreBackup(&restore);
```

Comment:
`aboutToRestoreBackup()` is never emitted in the new code, and `_hasForwardInTimeFiles/_backInTimeFiles` are set, but unused.

Reply:
True, this was done only if _account->serverVersionInt() < Account::makeServerVersion(9, 1, 0) ,  and this server is no longer supported.
So I'll remove the remains of it.